### PR TITLE
Add transactions queries

### LINKED ISSUE #240: Add transaction queries to query builder
Add the following queries to the query builder:
- `SHOW TRANSACTIONS;` -> https://memgraph.com/docs/memgraph/reference-guide/transactions#show-transactions
- `TERMINATE TRANSACTIONS "tid", "<tid2>", "<tid3>", ... ;` -> https://memgraph.com/docs/memgraph/reference-guide/transactions#terminate-transactions

### PR DESCRIPTION
**Pull request by katarinasupe**
### Description

I added get transactions and terminate transactions methods for easier transaction management.

### Pull request type

- [x] Feature

### Related issues

Closes #240 

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation TODO generate reference guide
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

######################################

### Reviewer checklist (the reviewer checks this part)
- [ ] Core feature implementation
- [ ] Tests
- [ ] Code documentation
- [ ] Documentation on [gqlalchemy/docs](https://github.com/memgraph/gqlalchemy/tree/main/docs)

######################################

### CODE REVIEW

File: gqlalchemy/vendors/memgraph.py [hunk 1]
```diff
@@ -432,3 +450,43 @@ def with_power_bi(self) -> "Memgraph":
...
+        query = (
+            "TERMINATE TRANSACTIONS " + ", ".join([f"'{transaction_id}'" for transaction_id in transaction_ids]) + ";"
+        )
+        print(query)
+        terminated_transactions = []
+        results = self.execute_and_fetch(query)
+
+        for result in results:
+            terminated_transactions.append(MemgraphTerminatedTransaction(result["transaction_id"], result["killed"]))
+        print(terminated_transactions)
```
Comment (antepusic) [inline gqlalchemy/vendors/memgraph.py:491]: Remove the debug comments:
```suggestion
        terminated_transactions = []
        results = self.execute_and_fetch(query)

        for result in results:
            terminated_transactions.append(MemgraphTerminatedTransaction(result["transaction_id"], result["killed"]))
```

**Review by antepusic — CHANGES_REQUESTED** [review]
I think this is all right once you apply the below suggestion:

**Comment by antepusic** [comment]
@katarinasupe Is this also milestone 1.6.0 like #309?

**Comment by katarinasupe** [comment]
> @katarinasupe Is this also milestone 1.6.0 like #309?

Yes! Thank you, I updated it. I just need to deal with the tests first since I bumped up Memgraph in the workflow :D Then I'll incorporate your suggestions.

File: tests/ogm/test_transactions.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+def test_get_transactions(memgraph):
+    result = memgraph.get_transactions()
+    assert len(result) == 1
```
Comment (as51340) [inline tests/ogm/test_transactions.py:3]: please add test for terminate transactions also

File: gqlalchemy/vendors/memgraph.py [hunk 3]
```diff
@@ -432,3 +452,43 @@ def with_power_bi(self) -> "Memgraph":
...
+        transactions = []
+
+        for transaction_data in transactions_data:
+            transaction = MemgraphTransaction(
+                username=transaction_data["username"],
+                transaction_id=transaction_data["transaction_id"],
+                query=transaction_data["query"],
+                metadata=transaction_data["metadata"],
+            )
+            transactions.append(transaction)
```
Comment (as51340) [inline gqlalchemy/vendors/memgraph.py:472]:
```suggestion
transactions_data = self.execute_and_fetch("SHOW TRANSACTIONS;")

# Define a helper function to create a MemgraphTransaction object from transaction data.
def create_transaction(transaction_data):
    return MemgraphTransaction(
        username=transaction_data["username"],
        transaction_id=transaction_data["transaction_id"],
        query=transaction_data["query"],
        metadata=transaction_data["metadata"],
    )

transactions = list(map(create_transaction, transactions_data))

```

File: gqlalchemy/vendors/memgraph.py [hunk 4]
```diff
@@ -432,3 +452,43 @@ def with_power_bi(self) -> "Memgraph":
...
+        query = (
+            "TERMINATE TRANSACTIONS " + ", ".join([f"'{transaction_id}'" for transaction_id in transaction_ids]) + ";"
+        )
+        print(query)
+        terminated_transactions = []
+        results = self.execute_and_fetch(query)
+
+        for result in results:
+            terminated_transactions.append(MemgraphTerminatedTransaction(result["transaction_id"], result["killed"]))
+        print(terminated_transactions)
```
Comment (as51340) [inline gqlalchemy/vendors/memgraph.py:493]: you can use map here in the same way

**Review by as51340 — COMMENTED** [review]
Nice work, few comments

File: gqlalchemy/vendors/memgraph.py [hunk 5]
```diff
@@ -432,3 +480,33 @@ def with_power_bi(self) -> "Memgraph":
...
+            transaction_ids (List[str]): A list of transaction ids to terminate.
+        Returns:
+            List[MemgraphTerminatedTransaction]: A list of MemgraphTerminatedTransaction objects with info on their status.
+        """
+
+        query = (
+            "TERMINATE TRANSACTIONS " + ", ".join([f"'{transaction_id}'" for transaction_id in transaction_ids]) + ";"
+        )
+
+        terminated_transactions = []
```
Comment (as51340) [inline gqlalchemy/vendors/memgraph.py:507]: delete this

**Review by as51340 — APPROVED** [review]
Just one small comment, otherwise good, I will approve. Delete unnecessary variable initialization

**Review by antepusic — APPROVED** [review]
Seconding Andi’s comment

**Pull request merged**